CalPatRec/src/AgnosticHelixFinder_module.cc

```diff
@@ -1083,7 +1077,7 @@ namespace mu2e {
           }
         }
       }
-      _tcHits[i].uselessTripletSeed = uselessSeed;
+      if (i<_tcHits.size()) {_tcHits[i].uselessTripletSeed = uselessSeed;}
```

Comment:
Just checking.  is it normal and correct behaviour for i to be >= _tcHits.size()?  Or is this test masking a deeper error?

Reply:
Yes. Just a handful on lines up (line 1067), I set i = _tcHits.size(). This is done because at that line in the for loop we have found a helix and i want to escape the i, j, and k loops right away. The helix is saved in the k loop, so i just break that one in the standard way. I am then forcing the j and i loops to end by setting i and j to be out of range. This is why line 1080 was having issues, prior to adding the if statement protection. Maybe it would be cleaner to create a bool, like "helixFound" or something, that is false by default, and then have break statement in the i/j loops when that bool is true. I can change that in my next PR, if you prefer.

Reply:
For a future PR.  There are two recommended ways to break out of a multiply nested control structure.  One is to use a goto statement.  The other is to put the nested structure in a function - then you can return from the function instead of using break.    Dave made an earlier comment about preferring more smaller functions vs fewer bigger functions.  This is one of the (many) reasons why.

Reply:
Issue #1270 has been created to follow up on  my comments and Dave's similar comments later in the review.